### PR TITLE
Clean up some of pkb.py

### PR DESCRIPTION
**Pull request by ehankland**

### CODE REVIEW

File: perfkitbenchmarker/benchmark_spec.py [hunk 1]
```diff
@@ -78,34 +80,43 @@ def UnPickleLock(locked, *args):
...
 
     Args:
-      benchmark_config: BenchmarkConfigSpec. The configuration for the
-          benchmark.
-      benchmark_name: string. Name of the benchmark.
+      benchmark_module: The benchmark module object.
       benchmark_uid: An identifier unique to this run of the benchmark even
           if the same benchmark is run multiple times with different configs.
-      spark_service: The spark service configured for this benchmark.
+      benchmark_config: BenchmarkConfigSpec. The configuration for the
```
Comment (yuyantingzero) [inline perfkitbenchmarker/benchmark_spec.py]: Follow order of arg list

File: perfkitbenchmarker/benchmark_spec.py [hunk 2]
```diff
@@ -403,18 +414,18 @@ def DeleteVm(self, vm):
...
-    config, self.config = self.config, None
-    with open(self.file_name, 'wb') as pickle_file:
+    with open(self._GetPickleFilename(self.uid), 'wb') as pickle_file:
       pickle.dump(self, pickle_file, 2)
-    self.config = config
 
   @classmethod
-  def GetSpecFromFile(cls, name, config):
+  def GetBenchmarkSpec(cls, benchmark_module, config, uid):
     """Unpickles the spec and returns it.
```
Comment (yuyantingzero) [inline perfkitbenchmarker/benchmark_spec.py]: Update docstring.

Comment (yuyantingzero) [inline perfkitbenchmarker/benchmark_spec.py]: Also arg list

Comment (ehankland) [inline perfkitbenchmarker/benchmark_spec.py]: Done

File: perfkitbenchmarker/benchmark_spec.py (hunk 1 above)
Comment (ehankland) [inline perfkitbenchmarker/benchmark_spec.py]: Done

**Pull request merged**